torch/csrc/jit/passes/integer_value_refinement.cpp

```diff
@@ -89,8 +89,10 @@ struct IntegerValueRefiner {
         }
         // one constant value one not - we are looking for the pattern
         // where y.1 is refined to the existing block output %one_constant
-        auto other_const_value =
-            constant_as<int64_t>(other_if_block->outputs().at(i));
+        auto other_output = other_if_block->outputs().at(i);
+        auto other_const_value = other_output->type()->cast<IntType>()
+            ? constant_as<int64_t>(other_output)
```

Comment:
wouldn't constant_as<**int64_t**> return a value iff the type is `IntType`?

Reply:
Unfortunately not, it throws an error if you run constant_as<int64_t> on a None constant. I actually had a PR to fix this and it was rejected lol.